### PR TITLE
Fix UB in FunctionArrayElement::executeGenericConst(...)

### LINKED ISSUE #19305: ArrayElement + ubsan
```sql
select [0][-9223372036854775808]
```
```
../src/Functions/array/arrayElement.cpp:234:55: runtime error: negation of -9223372036854775808 cannot be represented in type 'long'; cast to an unsigned type to negate this value to itself
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/Functions/array/arrayElement.cpp:234:55 in 
```

### PR DESCRIPTION
**Pull request by tavplubix**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Detailed description / Documentation draft:
https://clickhouse-test-reports.s3.yandex.net/19091/0507a38ec639e5eeba78f020ff2fc2d3d3447a0d/fuzzer_ubsan/report.html#fail1
Fixes #19305

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
@tavplubix I've made another fix in https://github.com/ClickHouse/ClickHouse/pull/19347. It looks more complete.

File: src/Functions/array/arrayElement.cpp [hunk 1]
```diff
@@ -588,7 +588,7 @@ ColumnPtr FunctionArrayElement::executeGenericConst(
             col_nested, col_array->getOffsets(), safeGet<UInt64>(index) - 1, *col_res, builder);
     else if (index.getType() == Field::Types::Int64)
         ArrayElementGenericImpl::vectorConst<true>(
-            col_nested, col_array->getOffsets(), -safeGet<Int64>(index) - 1, *col_res, builder);
+            col_nested, col_array->getOffsets(), -(safeGet<Int64>(index) + 1), *col_res, builder);
```
Comment (alexey-milovidov) [inline src/Functions/array/arrayElement.cpp:591]: It still has signed integer overflow when index is max positive Int64.

**Comment by alexey-milovidov** [comment]
Please review my PR then :)

File: src/Functions/array/arrayElement.cpp (hunk 1 above)
Comment (tavplubix) [reply src/Functions/array/arrayElement.cpp:591]: If I understand correctly, this branch is executed only when `index` is negative (we call `ArrayElementGenericImpl::vectorConst(...)` with `negative = true`), so there is no signed integer overflow.

**Pull request closed without merging**